project/profiles/capacity4more/modules/c4m/user/c4m_user_notifications/includes/c4m_user_notifications.helper.inc

```diff
@@ -106,9 +108,29 @@ function _c4m_user_notifications_message_subscribe_get_subscribers(Message $mess
       }
     }
 
-    // If user is found at filtering array, set it's notifier to mail.
+    // If user is found at filtering array, retrieve its notifier and set it.
     if (array_search($key, $filtered_users) !== FALSE) {
-      $user_object = ['notifiers' => ['email' => 'email']];
+      if (!module_exists('c4m_message_digest')) {
+        $user_object = ['notifiers' => ['email' => 'email']];
+      }
+      else {
+        $preference = c4m_message_digest_get_preference($flag->fid, $flagged_entity_type, $entity_id, $user_object->uid);
+
+        if (!empty($preference)) {
+          if ($preference->getPreference() === C4M_MESSAGE_DIGEST_FREQUENCY_NEVER) {
+            unset($users[$key]);
+            continue;
+          }
+          $user_object = ['notifiers' => [$preference->getPreference() => $preference->getPreference()]];
+        }
+        else {
+          // No specific preference; use immediate notification.
+//          $user_object = ['notifiers' => ['email' => 'email']];
+
+          unset($users[$key]);
+          continue;
+        }
+      }
```

Comment:
I think it's best to add this to a separate function. The current function is already big aprox 100 lines.